tests/baselines/reference/objectLiteralExcessProperties.errors.txt

```diff
@@ -91,4 +101,37 @@ tests/cases/compiler/objectLiteralExcessProperties.ts(33,27): error TS2322: Type
 !!! error TS2322: Type '{ colour: string; }' is not assignable to type 'Cover'.
 !!! error TS2322:   Object literal may only specify known properties, but 'colour' does not exist in type 'Cover'. Did you mean to write 'color'?
 !!! related TS6501 tests/cases/compiler/objectLiteralExcessProperties.ts:28:5: The expected type comes from this index signature.
+    
+    // Repros inspired by #28752
+    
+    function test<T extends IFoo>() {
+                            ~~~~
+!!! error TS2304: Cannot find name 'IFoo'.
+        // No excess property checks on generic types
+        const obj1: T = { name: "test" };
+              ~~~~
+!!! error TS2322: Type '{ name: string; }' is not assignable to type 'T'.
+        // No excess property checks on intersections involving generics
+        const obj2: T & { prop: boolean } = { name: "test", prop: true };
+              ~~~~
+!!! error TS2322: Type '{ name: string; prop: boolean; }' is not assignable to type 'T & { prop: boolean; }'.
+!!! error TS2322:   Type '{ name: string; prop: boolean; }' is not assignable to type 'T'.
+        // Excess property checks only on non-generic parts of unions
+        const obj3: T | { prop: boolean } = { name: "test", prop: true };
+                                              ~~~~~~~~~~~~

```

Comment:
This makes sense in an assignment context, but
```ts
declare function foo<T>(x: T | { prop: boolean }): T | boolean;

foo({prop: true, name: "test"}); // shouldn't be excess, trivially satisfies `T`
```
for calls it seems a little jank, perhaps?

Reply:
Does the excess error happen before inference of `T` to `{prop: true, name: string}`?

Reply:
@weswigham As @jack-williams points out, there's no error in your example because we infer and substitute a type for `T` before the assignment check.